src/SourceLocator/AutoloadSourceLocator.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace BetterReflection\SourceLocator;
+
+use BetterReflection\Identifier\Identifier;
+use BetterReflection\Identifier\IdentifierType;
+
+/**
+ * Use PHP's built in autoloader to locate a class, without actually loading.
+ *
+ * There are some prerequisites...
+ *   - we expect the autoloader to load classes from a file (i.e. using require/include)
+ */
+class AutoloadSourceLocator implements SourceLocator
+{
+    private static $autoloadLocatedFile;
+
+    public function __invoke(Identifier $identifier)
+    {
+        self::$autoloadLocatedFile = null;
+
+        $this->locateIdentifier($identifier);
+
+        if (null == self::$autoloadLocatedFile) {
+            throw new Exception\AutoloadFailure(sprintf(
+                'Unable to autoload the %s called %s',
+                $identifier->getType()->getName(),
+                $identifier->getName()
+            ));
+        }
+
+        return new LocatedSource(
+            file_get_contents(self::$autoloadLocatedFile),
+            self::$autoloadLocatedFile
+        );
+    }
+
+    /**
+     * Attempts to locate the specified identifier and store the result in
+     * self::$autoloadLocatedFile
+     *
+     * @param Identifier $identifier
+     */
+    private function locateIdentifier(Identifier $identifier)
+    {
+        if ($identifier->getType()->getName() == IdentifierType::IDENTIFIER_CLASS) {
```

Comment:
Cache the call to `$identifier->getType()->getName()`. You should also probably make a method that matches the constant, rather than getting the constant and then comparing outside the VO